Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/DirectoryInitialization.java

```diff
@@ -120,8 +120,13 @@ private static boolean setDolphinUserDirectory(Context context)
     Log.debug("[DirectoryInitialization] User Dir: " + userPath);
     NativeLibrary.SetUserDirectory(userPath);
 
-    File cacheDir = context.getExternalCacheDir();
-    if (cacheDir == null)
+    File cacheDir;
+
+    if (context.getExternalCacheDir() != null)
+      cacheDir = context.getExternalCacheDir();
+    else if (context.getCacheDir() != null)
+      cacheDir = context.getCacheDir();
+    else
       return false;
```

Comment:
Also, it would be useful to have a short comment explaining why we have an extra check. Normally it should be impossible for `getExternalCacheDir` to return null if `getExternalFilesDir` returns non-null. Something similar to what you currently have in the PR description should work.

Though, the PR description raises a new question for me... You say that it "occasionally" returns null. Is that something that can change at random while the app is running? Or only when the app is restarted? Or on even less common events?

Reply:
The way the if statement were ordered looked more sane at that moment but this works too. And my bad in the phrasing, it happens every time the function is called.